### PR TITLE
3.7 table relationships kyselov

### PR DESCRIPTION
**Pull request by IselDrew**
1. Add 'default' to models.
2. Change table fields naming.
3. Add relationships between Teacher-Group and Group-Student

Example of simple JOIN request (javascript): 
image

Result:
image

### CODE REVIEW

File: src/migrations/20200415231700-create-students-table.ts [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import { QueryInterface, DataTypes } from 'sequelize';
-import { Students } from '../modules/students/students.model';
+import Students from '../modules/students/students.model';
```
Comment (Alan18081) [inline src/migrations/20200415231700-create-students-table.ts:2]: Please, don't use default imports as much as possible

File: src/modules/groups/groups.model.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { DataTypes, Model, Sequelize } from 'sequelize';
+import { sequelize } from '../../database';
+import Teachers from '../teachers/teachers.model';
 
-export class Groups extends Model {
+export default class Groups extends Model {
```
Comment (Alan18081) [inline src/modules/groups/groups.model.ts:5]: Please, don't use default exports as much as possible

File: src/modules/students/students.model.ts [hunk 3]
```diff
@@ -1,43 +1,56 @@
 import { DataTypes, Model, Sequelize } from 'sequelize';
+import { sequelize } from '../../database';
+import Groups from '../groups/groups.model';
 
-export class Students extends Model {
+export default class Students extends Model {
+  public static readonly TableName: string = 'Students';
```
Comment (Alan18081) [inline src/modules/students/students.model.ts:6]: It should be lowercase

**Comment by antonpanchenk0** [comment]
pls, add "tableName: this.TableName" to Groups model options

File: src/modules/students/students.model.ts (hunk 3 above)
Comment (IselDrew) [reply src/modules/students/students.model.ts]: fixed

File: src/modules/groups/groups.model.ts (hunk 2 above)
Comment (IselDrew) [reply src/modules/groups/groups.model.ts]: fixed

File: src/migrations/20200415231700-create-students-table.ts (hunk 1 above)
Comment (IselDrew) [reply src/migrations/20200415231700-create-students-table.ts]: fixed

File: src/modules/teachers/teachers.model.ts [hunk 4]
```diff
@@ -31,7 +30,9 @@ export class Teachers extends Model {
       {
         sequelize: sequelize,
```
Comment (antonpanchenk0) [inline src/modules/teachers/teachers.model.ts:31]: tslint will give an error, you need to use es syntax. Just sequalize

File: src/modules/students/students.model.ts [hunk 5]
```diff
@@ -1,43 +1,56 @@
...
+        lastNameUkr: DataTypes.STRING(),
+        firstNameEng: DataTypes.STRING(),
+        lastNameEng: DataTypes.STRING(),
+        email: DataTypes.STRING(),
+        phoneNumber: DataTypes.STRING(),
+        password: DataTypes.STRING(),
+        groupId: DataTypes.INTEGER(),
+      },
+      {
+        sequelize: sequelize,
```
Comment (antonpanchenk0) [inline src/modules/students/students.model.ts:39]: tslint will give an error, you need to use es syntax. Just sequalize

File: src/modules/groups/groups.model.ts [hunk 6]
```diff
@@ -32,7 +34,22 @@ export class Groups extends Model {
           allowNull: true,
         },
       },
-      { sequelize }
+      {
+        sequelize: sequelize,
```
Comment (antonpanchenk0) [inline src/modules/groups/groups.model.ts:38]: tslint will give an error, you need to use es syntax. Just sequalize

File: src/migrations/20200415143345-create-groups.ts [hunk 7]
```diff
@@ -18,8 +19,13 @@ export async function up(query: QueryInterface) {
         allowNull: true,
       },
       teacherId: {
-        type: DataTypes.INTEGER(),
-        allowNull: true,
+        type: DataTypes.INTEGER,
```
[the remark this answers is not in the document]
Comment (IselDrew) [reply src/migrations/20200415143345-create-groups.ts:22]: My task requires rewriting of existing migrations. But ok, thanks, noted.

File: src/modules/students/students.model.ts [hunk 8]
```diff
@@ -1,43 +1,56 @@
...
-    public lastNameUkr: string;
-    public firstNameEng: string;
-    public lastNameEng: string;
-    public email: string;
-    public phoneNumber: string;
-    public password: string;
-    public groupId: number;
-    public createdAt: Date;
-    public updatedAt: Date;
+  public id!: number;
```
[the remark this answers is not in the document]
Comment (IselDrew) [reply src/modules/students/students.model.ts]: fixed

**Pull request merged**